### ISSUE TITLE
Python byndings for query.filter_<>() behaves strangly

### ISSUE THREAD

**Issue by j-mracek** [rpm-software-management/dnf5#1086]
Following code provides incorrect results - empty query
```
        for require in pkg.get_requires():
            query1 = libdnf.rpm.PackageQuery(query)
            query1.filter_provides([str(require)])
```
```
        for require in pkg.get_requires():
            query1 = libdnf.rpm.PackageQuery(query)
            query1.filter_provides(str(require))
```
```
            query1 = libdnf.rpm.PackageQuery(query)
            query1.filter_provides("dnf")
```

**Comment by j-mracek** [comment]
First problem
str(require) does return class instance description and not string representing reldep. The core requires to call `require.to_string()` to get required result.

**Comment by gotmax23** [comment]
Right, `libdnf5.rpm.Reldep` does not implement `__str__()`, so calling `str()` on it just returns the repr.

**Comment by j-mracek** [comment]
I create the first PR that resolves the issue partially - https://github.com/rpm-software-management/dnf5/pull/1092.

Remaining problem - passing string and not list of strings results in strange behavior (`query1.filter_provides("dnf")`).

**Comment by j-mracek** [comment]
It looks like that when string is passed `query1.filter_provides("dnf")` it is automatically converted to std::vector{"d", "n", "f"}. The problem is that that this incorrect conversion is not visible and hard to debug, because it returns some results time to time.

**Comment by j-mracek** [comment]
I've created two draft PR that represents alternative solutions.

[The first ](https://github.com/rpm-software-management/dnf5/pull/1101)  adds an additional overload method to `hpp`. This approach is potentially problematic, but nicer.


[The second](https://github.com/rpm-software-management/dnf5/pull/1101) adds the method only for Python binding.

To fix the issue completely it would require modifying a lot of methods, therefore I would like to know preferable way before implementation.

May I ask you for your opinion?

**Comment by m-blaha** [comment]
I personally prefer the additional overload. This makes the API the same across all languages.

**Comment by inknos** [comment]
> https://github.com/rpm-software-management/dnf5/pull/1101adds an additional overload method to hpp. This approach is potentially problematic, but nicer.
>
> https://github.com/rpm-software-management/dnf5/pull/1101 adds the method only for Python binding.

@j-mracek please, check the links, they both go to 1101, but I believe you wanted to go to 1102

**Comment by j-mracek** [comment]
The issue is fixed by https://github.com/rpm-software-management/dnf5/pull/1216
